capstone/src/test/java/com/google/sps/LessonTest.java

```diff
@@ -28,4 +49,150 @@ public void setUp() {
     public void tearDown() {
         helper.tearDown();
     }
-}
\ No newline at end of file
+
+    @Test
+    public void deserializeJsonFormTest() {
+        JsonObject jsonObject = new JsonObject();
+        jsonObject.addProperty(Lesson.TYPE_PROPERTY_KEY, Lesson.TYPE_FORM);
+        jsonObject.addProperty(Lesson.TITLE_PROPERTY_KEY, LessonTest.TEST_TITLE);
+        jsonObject.addProperty(Lesson.DESCRIPTION_PROPERTY_KEY, LessonTest.TEST_DESCRIPTION);
+        jsonObject.addProperty(Form.EDIT_URL_PROPERTY_KEY, LessonTest.TEST_EDIT_URL);
+        jsonObject.addProperty(Form.URL_PROPERTY_KEY, LessonTest.TEST_URL);
+
+        Lesson lesson = Lesson.deserializeJson(jsonObject.toString());
+
+        Assert.assertEquals(lesson.getType(), Lesson.TYPE_FORM);
+        Assert.assertEquals(lesson.getTitle(), LessonTest.TEST_TITLE);
+        Assert.assertEquals(lesson.getDescription(), LessonTest.TEST_DESCRIPTION);
+        Assert.assertEquals(lesson.getIsDraft(), false);
+        Assert.assertTrue(lesson instanceof Form);
+        Form form = (Form) lesson; 
+        Assert.assertEquals(form.getEditUrl(), LessonTest.TEST_EDIT_URL);
+        Assert.assertEquals(form.getUrl(), LessonTest.TEST_URL);
```

Comment:
Decided to type cast for each different type (video, lesson, form, content) so that if more fields are added to each one, we wouldn't have to type cast inline to call the method.